client-js/main/client/client.js

```diff
@@ -31,133 +36,141 @@
-   *                             or to be rejected with a `SpineError`
-   * @abstract
+   * @template <T> a Protobuf type of entities being the target of a query
    */
-  atOnce() {
+  execute(query) {
     throw new Error('Not implemented in abstract base.');
   }
-}
-
-/**
- * An abstract client for Spine application backend. This is a single channel for client-server
- * communication in a Spine-based browser application.
- *
- * @abstract
- */
-export class Client {
 
   /**
-   * Defines a fetch query of all entities matching the filters provided as arguments.
-   * This fetch is executed later upon calling the corresponding `.oneByOne()` and
-   * `.atOnce()` methods.
-   *
-   * `fetchAll(...).oneByOne()` queries the entities returning them in asynchronous manner using
-   * an observable. A subscriber is added to an observable to process each next entity or handle
-   * the error during the operation.
-   *
-   * `fetchAll(...).atOnce()` queries all the entities at once fulfilling a returned promise
-   * with an array of objects.
+   * Creates a new {@link TopicFactory} for building subscription topics specifying
+   * the state changes to be observed from Spine server.
    *
    * @example
-   * // Fetch all entities of a developer-defined Task type one-by-one using an observable.
-   * fetchAll({ofType: taskType}).oneByOne().subscribe({
-   *   next(task) { ... },
-   *   error(error) { ... },
-   *   complete() { ... }
-   * })
+   * // Build a subscription topic specifying all entities of a developer-defined `UserTasks` type
+   * newQuery().select(Task)
+   *           .build()
+   *
    * @example
-   * // Fetch all entities of a developer-defined Task type at once using a Promise.
-   * fetchAll({ofType: taskType}).atOnce().then(tasks => { ... })
+   * // Build a subscription topic specifying entities of a developer-defined `UserTasks`
+   * // type where tasks count is greater than 3
+   * newTopic().select(UserTasks)
+   *           .where(Filters.gt('tasksCount', 3))
+   *           .build()
    *
-   * @param {!Type<T>} ofType a type of the entities to be queried
-   * @return {Client.Fetch<T>} a fetch object allowing to specify additional remote
-   *                                call parameters and executed the query.
+   * To execute the resulting `Topic` instance pass it to the {@link Client#subscribeTo()}
+   * method.
    *
-   * @template <T>
+   * @return {TopicFactory} a factory for creating subscription topics to the Spine server
+   *
+   * @see TopicFactory
+   * @see TopicBuilder
+   * @see AbstractTargetBuilder
    */
-  fetchAll({ofType: type}) {
+  newTopic() {
     throw new Error('Not implemented in abstract base.');
   }
 
   /**
-   * Fetches a single entity of the given type.
+   * Creates a subscription to the topic which is updated with backend changes.
+   * Fulfills a returning promise with an object representing a result of the
```

Comment:
Let's not re-iterate what a subscription is over and over again. It should be shorter:

```
 Fulfills a returning promise with the created subscription.
```

Same below.